spring-integration-core/src/main/java/org/springframework/integration/config/annotation/InboundChannelAdapterAnnotationPostProcessor.java

```diff
@@ -59,7 +60,15 @@ public Object postProcess(Object bean, String beanName, Method method, List<Anno
 		String channelName = MessagingAnnotationUtils.resolveAttribute(annotations, AnnotationUtils.VALUE, String.class);
 		Assert.hasText(channelName, "The channel ('value' attribute of @InboundChannelAdapter) can't be empty.");
 
-		MessageSource<?> messageSource = this.createMessageSource(bean, beanName, method);
+		MessageSource<?> messageSource = null;
+		try {
+			messageSource = createMessageSource(bean, beanName, method);
+		}
+		catch (NoSuchBeanDefinitionException e) {
+			// Skip the @Bean for farther endpoint processing.
+			// Mainly by the @Conditional reason.
+			return null;
```

Comment:
I think we should add a DEBUG log here - just in case, in future, it's some other bean that's missing (not the `@Bean`).

It's ok in the abstract class because we're only looking for the `@Bean` there (instead of the more general `create...` method). However, it won't hurt to add a DEBUG (or TRACE) log there too.